docs/guide.md

```diff
@@ -0,0 +1,575 @@
+database using a familiar JavaScript-like language. The complete [language
+reference](language.md) describes the syntax. This guide introduces the concepts and
+features of Bolt along with a cookbook of common recipies.
+
+# Getting Started
+
+The Firebase Bolt compiler is a command-line based tool, written in node.js. You can install it
+using the [Node Package Manager](https://nodejs.org/en/download/):
+
+    $ npm install --global firebase-bolt
+
+You can use the Bolt compiler to compile the examples in this guide, and inspect the output
+JSON.
+
+## Default Firebase Permissions
+
+By default, Firebase has a permissive security model - this makes it easy to test your code
+(but is unsafe for production apps since anyone can read and overwrite all your data). In Bolt,
+these default permissions can be written as:
+
+[all_access.bolt](../samples/all_access.bolt)
+```javascript
+path / {
+  read() = true;
+  write() = true;
+}
+```
+
+This _path_ statement, defines read and write permissions to every part of your database
+(all paths beneath the root (`/`) of the database).  The `read` and `write` methods are defined
+using the syntax `read() = <expression>;`.  When the expression evaluates to `true`, reading
+(or writing) is allow at the given path location (and all children of that path).
+
+Use the Bolt compiler to convert this to Firebase JSON-formatted rules:
+
+    $ firebase-bolt < all_access.bolt
+
+```JSON
+{
+  "rules": {
+    ".read": "true",
+    ".write": "true"
+  }
+}
+```
+
+## How to Use Bolt in Your Application
+
+Bolt is (not yet) integrated into the online [Firebase Security and Rules
+Dashboard](https://www.firebase.com/account/). There are two ways to use Bolt to define rules
+for your application:
+
+1. Use the firebase-bolt command line tool to generate a JSON file from your Bolt file, and
+   then copy and paste the result into the Dashboard _Security and Rules_ section.
+2. Use the [Firebase Command Line](https://www.firebase.com/docs/hosting/command-line-tool.html)
+   tool.  If you have _firebase-bolt_ installed on your computer, you can set the `rules` property
+   in your [firebase.json](https://www.firebase.com/docs/hosting/guide/full-config.html) file
+   to the name of your Bolt file.  When you use the `deploy` command, the command line will
+   read and compile your Bolt file and upload the compiled JSON to your application Rules.
+
+## Data Validation
+
+The Firebase database is "schemaless" - which means that, unless you specify otherwise, any
+type or structure of data can be written anywhere in the database. By specifying a specific
+schema, you can catch coding errors early, and prevent malicious programs from writing data
+that you don't expect.
+
+Lets say your application wants to allow users to write messages to your database. Each message
+can be up to 140 characters and must by _signed_ by the user. In Bolt, you can express this
```

Comment:
What does "signed by" mean here? If it is the `from` property, I would just say "and we'll also store who posted each message".